### DISCUSSION TITLE
Maximum speed which recorded in the track is not a real

### DISCUSSION THREAD

**Discussion by eremitein** [OpenTracksApp/OpenTracks#978] [Q&A] [answered]
Hi, thanks for this great app! 

I have installed OpenTracks v3.20.2 on Android 11 and here is one weird problem - maximum speed which recorded in the track is not a real maximum speed which I reached while riding. For sample, while riding I see current speed over 30 km/h but in the max speed indicates only 24 km/h. Also sometimes value of max speed was decreased (why?O_o)

In another app which i used before there is no such issue, maximum speed is always stored properly in the statistics 

But i like OpenTracks app and want using it, so please can someone help me - is it some option which i need change in the settings?

**Comment by eremitein** [comment]
Here is a screenshots. As you can see, Max speed is not relevant to the real values

![photo_2021-10-12_22-40-36](https://user-images.githubusercontent.com/12787224/137003684-54c8b5c3-18ff-4155-97b4-ec920c6f91e8.jpg)
![photo_2021-10-12_22-40-31](https://user-images.githubusercontent.com/12787224/137003691-cca97b22-e2a0-4638-a703-34c9081a9d7a.jpg)
![photo_2021-10-12_22-40-23](https://user-images.githubusercontent.com/12787224/137003694-845068d7-ffd3-4085-bea6-7d17c2ec57b8.jpg)

**Comment by dennisguse** [comment, marked as answer]
OpenTracks stores the speed data reported by the GPS (so it can be exported later).
To compute the statistics, the data is filtered to account for unreliable GPS data (e.g., due to bad reception etc).
Thus, the currently shown one might be above the max speed.

I am not really happy with the filtering as it does not take the time between GPS points into account.
And we even apply this filter if we get speed from BLE sensors - and I am even less happy about this.

**Reply by eremitein** [reply]
Well, can you rework code logic for max speed calculation? This patch need only for "live" UI layout rendering part (_sorry, i have no ideas about post processing and filters workaround_). Something like the simple algorithm:
(create loop 100ms and put inside)
take value from current speed (Scur) and store it as a value max speed (Smax)
if next iteration Scur > Smax then Smax = Scur else skip

I will be happy to check this debug build if you will created it


BTW I did a test today, did upload GPX tracks to the Strava. And, you know, _seems_ Strava shows right max speed (_i'm still not sure about Strava accuracy_)

**Reply by dennisguse** [reply]
Then Strava just does the computation differently.
Can you just create an issue for this?
It is definitely something, we can improve.

PS/as you cycle: you may consider BLE sensors as those are GPS reception independent and usually speed is more accurate (aka varies less).

**Comment by edern-appere** [comment]
I've noticed the same bug during my last couple of bike rides. It's especialy frustrating to go pretty fast downhill (OT showing 57 km/h as current speed) but at the end of the trip the max speed displayed in the overview is something like 30 km/h.

I don't know if it's related but I often see that speed in OT changes very quickly and by a big margin whereas I do not feel like my "real" speed changes so much. For example, when I'm on a long steady climb sometimes speed can change from 17 to 9 back to 17 all in 1 or 2 seconds...
